mcproto/types/nbt.py

```diff
@@ -208,7 +208,7 @@ def serialize(self, with_type: bool = True, with_name: bool = True) -> Buffer:
 
     @override
     @classmethod
-    def deserialize(cls, buf: Buffer, with_name: bool = True, with_type: bool = True) -> NBTag:
+    def deserialize(cls, buf: Buffer, with_name: bool = True, with_type: bool = True) -> Self:
```

Comment:
The `Self` annotation here seems wrong. This will just return the associated nbt tag class, that's not necessarily the instance of the caller class. If that's the intention, add an `isinstance` check, if not, why the `Self` type?